src/plus/ai/aiProviderService.ts

```diff
@@ -325,12 +325,18 @@ export class AIProviderService implements Disposable {
 
 		let chosenProviderId: AIProviders | undefined;
 		let chosenModel: AIModel | undefined;
+		const orgAiConf = getOrgAIConfig();
 
 		if (!options?.force) {
-			const vsCodeModels = await this.getModels('vscode');
-			if (vsCodeModels.length !== 0) {
-				chosenProviderId = 'vscode';
-			} else if ((await this.container.subscription.getSubscription()).account?.verified) {
+			if (isProviderEnabledByOrg('vscode', orgAiConf)) {
+				const vsCodeModels = await this.getModels('vscode');
+				if (vsCodeModels.length !== 0) {
+					chosenProviderId = 'vscode';
+				}
+			} else if (
```

Comment:
This changes the previous logic such that if we do not find a vscode model, it does not move to the `else` block and therefore does not choose the GK model. I suggest moving the existing vscode models check into the `if` statement to make sure that the original flow is preserved.

i.e.
```suggestion
                        const vsCodeModels = await this.getModels('vscode');
			if (isProviderEnabledByOrg('vscode', orgAiConf) && vsCodeModels.length !== 0) {
					chosenProviderId = 'vscode';
			} else if (
```